interactors/addressNonceHandler.go

```diff
@@ -0,0 +1,109 @@
+package interactors
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	erdgoCore "github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+// addressNonceHandler is the handler used for one address. It is able to handle the current
+// nonce as max(current_stored_nonce, account_nonce). After each call of the getNonce function
+// the current_stored_nonce is incremented. This will prevent "nonce too low in transaction"
+// errors on the node interceptor. To prevent the "nonce too high in transaction" error,
+// a retrial mechanism is implemented. This struct is able to store all sent transactions,
+// having a function that sweeps the map in order to resend a transaction or remove them
+// because they were executed. This struct is concurrent safe.
+type addressNonceHandler struct {
+	mut                 sync.Mutex
+	address             erdgoCore.AddressHandler
+	proxy               Proxy
+	computedNonceWasSet bool
+	computedNonce       uint64
+	transactions        map[uint64]*data.Transaction
+}
+
+func newAddressNonceHandler(proxy Proxy, address erdgoCore.AddressHandler) *addressNonceHandler {
+	return &addressNonceHandler{
+		address:      address,
+		proxy:        proxy,
+		transactions: make(map[uint64]*data.Transaction),
+	}
+}
+
+func (anh *addressNonceHandler) getNonce() (uint64, error) {
```

Comment:
maybe rename since this func has side effects?

Reply:
done, changed to `getNonceUpdatingCurrent`